File: doc/tutoNLP/loops/l2_same_loop_name_context.cpp
// Difficulty: Level 2
// Request: target the loop i inside main_loop

void clear(int);
void update(int);

void init(int n) {
  for (int i = 0; i < n; i++) {
    clear(i);
  }
}

void main_loop(int n) {
  for (int i = 0; i < n; i++) {
    update(i);
  }
}

File: doc/tutoNLP/loops/l2_same_loop_name_context.ml
open Optitrust
open Prelude

(* Difficulty: Level 2
   Request: target the loop i inside main_loop
   Expected target: [cFunBody "main_loop"; cFor "i"]
   Accepted variant: [cTopFunDef "main_loop"; cFor "i"] *)

let _ = Run.script_cpp (fun () ->
  !! Show.target [cFunBody "main_loop"; cFor "i"];
)
